src/main/java/software/amazon/event/kafkaconnector/mapping/JsonMapper.java

```diff
@@ -0,0 +1,89 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector.mapping;
+
+import static java.util.Collections.singletonMap;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import java.io.IOException;
+import org.apache.kafka.connect.header.Header;
+import org.apache.kafka.connect.json.JsonConverter;
+import org.apache.kafka.connect.sink.SinkRecord;
+
+public class JsonMapper {
+  public final JsonConverter jsonConverter = new JsonConverter();
+  public final ObjectMapper objectMapper = new ObjectMapper();
+
+  public JsonMapper() {
+    jsonConverter.configure(singletonMap("schemas.enable", "false"), false);
+  }
+
+  public String createJsonPayload(SinkRecord sinkRecord) throws IOException {
+    var root = objectMapper.createObjectNode();
+    root.put("topic", sinkRecord.topic());
+    root.put("partition", sinkRecord.kafkaPartition());
+    root.put("offset", sinkRecord.kafkaOffset());
+    root.put("timestamp", sinkRecord.timestamp());
+    root.put("timestampType", sinkRecord.timestampType().toString());
+    root.set("headers", createHeaderArray(sinkRecord));
+
+    if (sinkRecord.key() == null) {
+      root.set("key", null);
+    } else {
+      root.set(
+          "key",
+          createJSONFromByteArray(
+              jsonConverter.fromConnectData(
+                  sinkRecord.topic(), sinkRecord.keySchema(), sinkRecord.key())));
+    }
+
+    // tombstone handling
+    if (sinkRecord.value() == null) {
+      root.set("value", null);
+    } else {
+      root.set(
+          "value",
+          createJSONFromByteArray(
+              jsonConverter.fromConnectData(
+                  sinkRecord.topic(), sinkRecord.valueSchema(), sinkRecord.value())));
+    }
+    return root.toString();
+  }
+
+  /**
+   * This method serializes Kafka message headers to JSON.
+   *
+   * @param sinkRecord Kafka record to be sent to EventBridge
+   * @return headers to be added to EventBridge message
+   * @throws IOException
+   */
+  private ArrayNode createHeaderArray(SinkRecord sinkRecord) throws IOException {
+    var headersArray = objectMapper.createArrayNode();
+
+    for (Header header : sinkRecord.headers()) {
+      var headerItem = objectMapper.createObjectNode();
+      headerItem.set(
+          header.key(),
+          createJSONFromByteArray(
+              jsonConverter.fromConnectHeader(
+                  sinkRecord.topic(), header.key(), header.schema(), header.value())));
+      headersArray.add(headerItem);
+    }
+    return headersArray;
+  }
+
+  /**
+   * This method converts the byteArray which is returned by the {@link JsonConverter} to JSON.
+   *
+   * @param jsonBytes - byteArray to convert to JSON
+   * @return the JSON representation of jsonBytes
+   * @throws IOException
+   */
+  public JsonNode createJSONFromByteArray(byte[] jsonBytes) throws IOException {
```

Comment:
Can be private

Reply:
We use the method "createJSONFromByteArray" in our TimeMapper interface. I t would be nice to have it available, as dicussed in #358 .

Reply:
See above comment